bot/exts/info/python_news.py

```diff
@@ -139,26 +118,31 @@ async def post_pep_news(self) -> None:
                 avatar_url=AVATAR_URL,
                 wait=True,
             )
-            payload["data"]["pep"].append(pep_nr)
 
-            # Increase overall PEP new stat
-            self.bot.stats.incr("python_news.posted.pep")
-
-            if msg.channel.is_news():
-                log.trace("Publishing PEP announcement because it was in a news channel")
-                await msg.publish()
+            try:
+                # Update the seen peps in DB
+                await self.bot.api_client.post("bot/mailing-lists/pep/seen-items", json=pep_nr)
+                # Increase overall PEP new stat
+                self.bot.stats.incr("python_news.posted.pep")
+                self.seen_items["pep"].add(pep_nr)
 
-        # Apply new sent news to DB to avoid duplicate sending
-        await self.bot.api_client.put("bot/bot-settings/news", json=payload)
+                if msg.channel.is_news():
+                    log.trace("Publishing PEP announcement because it was in a news channel")
+                    await msg.publish()
+            except ResponseCodeError as e:
+                non_field_errors = e.response_json.get("non_field_errors", [])
+                if non_field_errors == ["Seen item already known."]:
+                    continue
+                raise e
```

Comment:
This is kind of duplicated with the other block below. Can we turn it into a function? We could update the log.trace call to contain the mailing list name, so you could imply it's a PEP for those.

Reply:
Available in b93972d7644c42637f1e985c979bb8de77f7c352

I didn't want to add the message publishing thing because it's out of scope of the method.